src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -491,7 +492,21 @@ protected static void ValidateExpandedNestedResourceInfoPropertyValue(IJsonReade
                 // Expanded resource (null or non-null)
                 if (isCollection == true)
                 {
-                    throw new ODataException(ODataErrorStrings.ODataJsonLightResourceDeserializer_CannotReadCollectionNestedResource(nodeType, propertyName));
+                    if (typeReference.IsODataPrimitiveTypeKind() ||
+                        typeReference.IsODataTypeDefinitionTypeKind() ||
+                        typeReference.IsODataEnumTypeKind() ||
+                        typeReference.IsODataComplexTypeKind() ||
+                        typeReference.IsUntyped() ||
+                        typeReference.IsNonEntityCollectionType())

```

Comment:
All these is either a structured type or just an edm type definition. Any chance the checks could be compressed?

Reply:
+1 

what is the semantic you're checking for here? is there an easier way to check?

---
In reply to: [439973000](https://github.com/OData/odata.net/pull/1793#discussion_r439973000) [](ancestors = 439973000)

Reply:
resolved